### PR TITLE
Move planner from STM32 to Raspberry Pi

### PR DESCRIPTION
**Pull request by ecourtois**

### CODE REVIEW

File: lib/cogip_defs/include/cogip_defs/Pose.hpp [hunk 1]
```diff
@@ -63,7 +63,7 @@ class Pose : public Coords {
         pose.set_O(O_);
     };
 
-private:
+protected:
```
Comment (gdoffe) [inline lib/cogip_defs/include/cogip_defs/Pose.hpp:66]: Why this change ?

File: lib/path/Pose.cpp [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+    if (max_speed_angular_ == LOW_SPEED_ANGULAR) {
+        path_pose.set_max_speed_angular_enum(PB_SpeedEnum::LOW);
+    }
+    else if (max_speed_angular_ == MAX_SPEED_ANGULAR) {
+        path_pose.set_max_speed_angular_enum(PB_SpeedEnum::MAX);
+    }
+    else {
+        path_pose.set_max_speed_angular_enum(PB_SpeedEnum::NORMAL);
+    }
```
Comment (gdoffe) [inline lib/path/Pose.cpp:84]: not a switch/case fan ? xD

File: lib/path/include/path/Pose.hpp [hunk 3]
```diff
@@ -52,6 +54,16 @@ class Pose : public cogip_defs::Pose {
     /// Execute action callback if set.
     virtual void act() const { if (act_) act_(); }
 
+    /// Initialize the object from a protobuf message.
```
Comment (gdoffe) [inline lib/path/include/path/Pose.hpp:57]: Protobuf

File: controllers/common_controller/include/ctrl.hpp [hunk 4]
```diff
@@ -60,6 +60,12 @@
 /* Project includes */
 #include "odometry.hpp"
 
+#ifdef MODULE_UARTPB
+#include "uartpb/UartProtobuf.hpp"
+
+inline constexpr cogip::uartpb::uuid_t pose_reached_uuid = 2736246403;  ///< pose reached uuid for uartpb
+#endif
```
Comment (gdoffe) [inline controllers/common_controller/include/ctrl.hpp:67]: Is it an UUID per message or per module ?

Comment (gdoffe) [inline controllers/common_controller/include/ctrl.hpp:67]: `#endif // MODULE_UARTPB`

**Review by gdoffe — COMMENTED** [review]
LGTM, more RAM, more flash available for new conrollers !
Just few comments/questions before approval.

File: lib/path/include/path/Pose.hpp (hunk 3 above)
Comment (ecourtois) [reply lib/path/include/path/Pose.hpp:57]: Done

File: controllers/common_controller/include/ctrl.hpp (hunk 4 above)
Comment (ecourtois) [reply controllers/common_controller/include/ctrl.hpp:67]: Per message, and a module can handle multiple messages.

File: lib/cogip_defs/include/cogip_defs/Pose.hpp (hunk 1 above)
Comment (ecourtois) [reply lib/cogip_defs/include/cogip_defs/Pose.hpp:66]: Because it is used if path::Pose, but it is not in the related commit.
Fixed.

File: lib/path/Pose.cpp (hunk 2 above)
Comment (ecourtois) [reply lib/path/Pose.cpp:84]: You cannot use a `switch/case` on floating-point values...

File: controllers/common_controller/include/ctrl.hpp (hunk 4 above)
Comment (ecourtois) [reply controllers/common_controller/include/ctrl.hpp]: Done

Comment (gdoffe) [reply controllers/common_controller/include/ctrl.hpp]: :+1:

**Review by gdoffe — APPROVED** [review]
LGTM!
